common/string_helpers.h

```diff
@@ -32,6 +34,38 @@ auto ParseBlockStringLiteral(llvm::StringRef source, int hashtag_num = 0)
 // correctness.
 auto StringRefContainsPointer(llvm::StringRef ref, const char* ptr) -> bool;
 
+// Translates `args` into C-string arguments for APIs based on `main`-like
+// command line argument lists.
```

Comment:
I think this would be a little clearer for me if it focused more on the contract rather than what it's used for, e.g.

```suggestion
// Converts `tool_path` and the contents of `args` to nul-terminated strings,
// and returns the results. This is particularly intended for use with APIs
// that expect `argv`-like command line argument lists.
```

(Changing "C-string" -> "nul-terminated string" and "`main`-like" -> "`argv`-like" are minor tweaks that would also have helped me a bit; feel free to take or leave them.)

Reply:
I prefer "C-string" as I think it's a more clear term, and more consistently spelled. But otherwise, sure?